src/lib/util/state.ts

```diff
@@ -92,6 +92,17 @@ export const updateConfig = (config: string, updateEditor: boolean): void => {
 	});
 };
 
+export const toggleDarkTheme = (dark: boolean): void => {
+	codeStore.update((state) => {
+		const config = JSON.parse(state.mermaid);
+		if (!config.theme || ['dark', 'default'].includes(config.theme)) {
+			config.theme = dark ? 'dark' : 'default';
+		}
+
+		return { ...state, mermaid: JSON.stringify(config, null, 2), updateEditor: true };
```

Comment:
Don't format JSON in production.

```suggestion
		return { ...state, mermaid: JSON.stringify(config), updateEditor: true };
```

Reply:
This JSON is the one shown in the config tab. So it needs to be formatted.